src/activity-logger/background/index.js

```diff
@@ -9,3 +9,68 @@ browser.webNavigation.onCommitted.addListener(({url, tabId, frameId}) => {
     maybeLogPageVisit({url, tabId})
 
 })
+
+
+//********************************************************************************/
+var timestamp = new Date().getTime;
+
+var number_of_opened_tabs = 0;
+
+var tabList = [];
+
+
+/*chrome.runtime.onMessage.addListener(function(response,sender,sendResponse){
+    alert(response);
+});*/
+
+
+browser.tabs.onCreated.addListener(function (tab){
+    number_of_opened_tabs = number_of_opened_tabs + 1;
+    //console.log('total opened tabs: ' + number_of_opened_tabs);
+    var t = {
+        id : tab.id,
+        activetime : 0,
+        isActive : false,
+        lastActivatedtime : Number(new Date().getTime())
+    };
+    tabList.push(t);
+    console.log('a new Tab created : ' + tab.id);
+})
+
+browser.tabs.onRemoved.addListener(function (id){
+    number_of_opened_tabs = number_of_opened_tabs - 1;
+    //console.log('total opened tabs: ' + number_of_opened_tabs);
+    for (var index in tabList){
+        if(tabList[index].id == id){
+            tabList.splice( index, 1 );
+        }
+    }
+    console.log('a Tab was closed : ' + id);
+})
+
+browser.tabs.onActivated.addListener(function (tab){
+    //console.log('Tab activated : ')
+    //console.log(tab);
+    for (var index in tabList){
+        if(tabList[index].isActive == true){
+            tabList[index].isActive = false;
+            tabList[index].lastActivatedtime = Number(new Date().getTime());
+            console.log('a tab went inactive : ' + tabList[index].id);
+        }
+        
+    }
+    for (var index in tabList){
```

Comment:
I want second loop work to be run after the first loop work. 
is there anything I can do for that? I think probably including a hashmap might reduce the complexity